packages/db/prisma/schema.prisma

```diff
@@ -77,10 +77,10 @@ model User {
     payments                 Payment[]
     preference               Preference?
     membership               Membership  @default(FREE)
-    paidUntil                Int? // in milliseconds
+    paidUntil                DateTime?
```

Comment:
It'd be nice with some context on what these fields represent and figure out what we need. Seems like we're saving a bunch of dates now and many will be duplicates.

cancelledDate to me is when the user decides to cancel their subscription, i.e. should be set to the current time.

then we have when their membership ends which will be at the end of the payment cycle

@bgavas - Right now the models are basically copied over from the backend repo, should we do some modifications?

Reply:
Hey Julius, I am not sure if modification is necessary. It was working fine with current fields. But @p6l-richard might want to have several changes.

`paidUntil` shows until when the user has subscription. Everytime the subscription is renewed this field is increased by a value of 1 month or 12 months based on the subscription.

`cancelledDate` shows when the user canceled the active subscription. With this field, we can update the `Next Payment` text in the screenshot with `Expires On`

<img width="253" alt="Screen Shot 2022-11-19 at 10 13 12" src="https://user-images.githubusercontent.com/30385360/202839464-3bb6fcd7-6639-468a-b5ce-6429e873a7cb.png">

This field is not updated when the subscription ends I believe. We don't have that value. Basically, paidUntil is the end of the subscription

Reply:
@juliusmarminge for context, we're currently not handling the trial via Stripe.

I personally prefer to have it be handled via stripe so that anything that concerns payments & billing, stripe becomes our source of truth for that and we simply copy the stripe objects that we feel we need in our database.

E.g. we can store the Stripe Customer and Subscription objects in our database and also log the stripe events for easier access -- this is a point that @bgavas made, which I agree on.

That way, we don't have to store some custom dates anywhere and can just refer to stripe objects defined by their api (following also their naming convention for consistency).

I searched around a bit and found this [repo here](https://github.com/nramkissoon/t3-stripe) which seems to implement a majority of the things I discussed above, except for the proper stripe objects. There's [an article](https://blog.nickramkissoon.com/posts/integrate-stripe-t3) for this setup, too.


What's your take on these two approaches (current copy-pasted from backend & the one outlined above), @juliusmarminge ?

Reply:
> I personally prefer to have it be handled via stripe so that anything that concerns payments & billing, stripe becomes our source of truth for that and we simply copy the stripe objects that we feel we need in our database.

Yes that sounds reasonable.

> E.g. we can store the Stripe Customer and Subscription objects in our database and also log the stripe events for easier access -- this is a point that @bgavas made, which I agree on.

Sure, so you want to store everything we get back from Stripe?

> I searched around a bit and found this [repo here](https://github.com/nramkissoon/t3-stripe) which seems to implement a majority of the things I discussed above, except for the proper stripe objects. There's [an article](https://blog.nickramkissoon.com/posts/integrate-stripe-t3) for this setup, too.

I don't see much difference with this and the current setup tbf

Reply:
> > E.g. we can store the Stripe Customer and Subscription objects in our database and also log the stripe events for easier access -- this is a point that @bgavas made, which I agree on.
> Sure, so you want to store everything we get back from Stripe?

Yes 👍 


> > I searched around a bit and found this [repo here](https://github.com/nramkissoon/t3-stripe) which seems to implement a majority of the things I discussed above, except for the proper stripe objects. There's [an article](https://blog.nickramkissoon.com/posts/integrate-stripe-t3) for this setup, too. 
> I don't see much difference with this and the current setup tbf

The differences I see are:
- subscription trial managed via stripe instead of via custom logic through `paidUntil` and some other date timestamps
- following stripe's API + naming convention in our database models

Reply:
> paidUntil shows until when the user has subscription. Everytime the subscription is renewed this field is increased by a value of 1 month or 12 months based on the subscription.
>
> cancelledDate shows when the user canceled the active subscription. With this field, we can update the Next Payment text in the screenshot with Expires On

You can just use the Stripe Billing Portal for this, where the user has more fine grained control over what to do. Or do you wan't to have your own tab for this?

<img width="1433" alt="CleanShot 2022-11-21 at 09 35 09@2x" src="https://user-images.githubusercontent.com/51714798/203003614-d63837af-f197-4814-ab05-05a8b381220b.png">

Reply:
Hi @juliusmarminge, I talked to Richard and we agreed on not having a dedicated tab for this. Customer portal can be handled by Stripe

Reply:
Alright so we'll just store the subscription object and use joins to get the dates or still want duplicates on the user object?

Reply:
Yes, sounds good to me -- the join shouldn't be too costly right now.